packages/webview_flutter/webview_flutter_platform_interface/lib/src/v4/webview_controller_delegate.dart

```diff
@@ -0,0 +1,264 @@
+  }
+
+  /// Runs the given JavaScript in the context of the current page.
+  ///
+  /// The Future completes with an error if a JavaScript error occurred.
+  Future<void> runJavaScript(String javascript) {
+    throw UnimplementedError(
+        'WebView runJavaScript is not implemented on the current platform');
+  }
+
+  /// Runs the given JavaScript in the context of the current page, and returns the result.
+  ///
+  /// The Future completes with an error if a JavaScript error occurred, or if the
+  /// type the given expression evaluates to is unsupported. Unsupported values include
+  /// certain non-primitive types on iOS, as well as `undefined` or `null` on iOS 14+.
+  Future<String> runJavaScriptReturningResult(String javascript) {
+    throw UnimplementedError(
+        'WebView runJavaScriptReturningResult is not implemented on the current platform');
+  }
+
+  /// Adds new JavaScript channels to the set of enabled channels.
+  ///
+  /// For each value in this list the platform's webview should make sure that a corresponding
+  /// property with a postMessage method is set on `window`. For example for a JavaScript channel
+  /// named `Foo` it should be possible for JavaScript code executing in the webview to do
+  ///
+  /// ```javascript
+  /// Foo.postMessage('hello');
+  /// ```
+  ///
+  /// See also: [CreationParams.javascriptChannelNames].
+  Future<void> addJavaScriptChannels(Set<String> javascriptChannelNames) {
+    throw UnimplementedError(
+        'WebView addJavaScriptChannels is not implemented on the current platform');
+  }
+
+  /// Removes JavaScript channel names from the set of enabled channels.
+  ///
+  /// This disables channels that were previously enabled by [addJavascriptChannels] or through
+  /// [CreationParams.javascriptChannelNames].
+  Future<void> removeJavaScriptChannels(Set<String> javascriptChannelNames) {
+    throw UnimplementedError(
+        'WebView removeJavaScriptChannels is not implemented on the current platform');
+  }
+
+  /// Returns the title of the currently loaded page.
+  Future<String?> getTitle() {
+    throw UnimplementedError(
+        'WebView getTitle is not implemented on the current platform');
+  }
+
+  /// Set the scrolled position of this view.
+  ///
+  /// The parameters `x` and `y` specify the position to scroll to in WebView pixels.
+  Future<void> scrollTo(int x, int y) {
+    throw UnimplementedError(
+        'WebView scrollTo is not implemented on the current platform');
+  }
+
+  /// Move the scrolled position of this view.
+  ///
+  /// The parameters `x` and `y` specify the amount of WebView pixels to scroll by.
+  Future<void> scrollBy(int x, int y) {
+    throw UnimplementedError(
+        'WebView scrollBy is not implemented on the current platform');
+  }
+
+  /// Return the horizontal scroll position of this view.
+  ///
+  /// Scroll position is measured from left.
+  Future<int> getScrollX() {
```

Comment:
For the new interface, we should replace `getScrollX` and `getScrollY` with `Future<Point<int>> getScrollPosition`. Since the methods are async, we realized its better that we retrieve their values together to ensure the (x, y) points are accurate.

Reply:
Makes sense, applied a fix for this.